src/drafts/MarkdownEditor/_useListEditing.ts

```diff
@@ -60,9 +60,9 @@ export const parseListItem = (line: string): ListItem | null => {
 }
 
 export const listItemToString = (item: ListItem) =>
-  `${item.leadingWhitespace}${typeof item.delimeter === 'number' ? `${item.delimeter}.` : item.delimeter}${
-    item.middleWhitespace
-  }${item.taskBox || ''} ${item.text}`
+  typeof item.delimeter === 'number'
```

Comment:
I've split this on 2 lines for readability + different handling : in the case of numbered list, i chose to ignore the "middle" spacing and format the MD correctly - because it is displayed in the correct way, ignoring the extra spaces

Reply:
Not sure I totally understand this. Aren't those extra spaces always displayed as a single space? I figure we'd want to keep the user's raw markdown as close as possible to what they actually wrote when we modify it.

Reply:
Yes, this changes the number of spaces for lists that are numbered. I intentionally did this - not sure it's correct but it was a subjective thinking that lists with numbers , given they're also displayed with single space, might as well be "fixed"

Reply:
Removed the intentional "fixes" and added tests [here](https://github.com/primer/react/pull/3987/files#diff-eeb126057e3f685f10b98cb5996d314f210a377d8a829c27b6481fe792ac92abR18) and [here](https://github.com/primer/react/pull/3987/files#diff-eeb126057e3f685f10b98cb5996d314f210a377d8a829c27b6481fe792ac92abR77)